website/docs/reference/widgets/list.md

```diff
@@ -41,70 +35,110 @@ You can display static JSON data in the `Items` property for generating the list
 
 ```javascript
-
 {{GetAllEmployees.data}}
 ```
 
 To learn how to bind data from JS functions, see [Display Data from JS function](/core-concepts/writing-code/workflows#display-data-from-async-js-function)
 
-If you are binding the dynamic data to the List widget, remember to use the  columns/attribute name to map to the individual widget as shown below:
+To display the data in individual widgets in the list item cards, use the **currentItem** property to bind the corresponding value from the object's fields in the widget as shown below. 
  
 ```javascript
-
 {{currentItem.<attribute_or_column_name>}}
 ```
 
+where `currentItem` for the first list item reflects the 0th object in the dataset. This can be used anywhere within a widget that's placed inside the List widget.
+
+
+### Unique list item identifier
+
+The List widget needs to identify each item uniquely to update, reorder, add or remove them. Similar to the `Primary Key` concept in the database or `key` in React, an identifier should be selected from the **Data identifier** property dropdown whose values are unique in the dataset provided to the List widget. Always set the **Data Identifier** property with a valid unique identifier to boost performance.
+
+In the preceding example, the identifier `bookId` has a unique value in the dataset.
+
+If no such unique identifier is present in the data, you can join multiple identifiers  to form a unique pattern by enabling the `JS` mode in the property.
+
+**Example:**
+```
+{{currentItem.bookName + "_" + currentItem.author}}
+```
+
 ## Server-side pagination
 
-Lists are often required to display large data sets from queries, but browsers can only sometimes load all the data present in the database or might do so slowly. You can use server-side pagination when a client receives only a subset of data from large datasets. It allows you to define the data limit that a query call can render. Thus, allowing you to paginate the data and determine the pagination boundaries.
+Lists are often required to display large data sets from queries, but browsers can only sometimes load all the data in the database or might do so slowly. You can use server-side pagination when a client receives only a subset of data from large datasets. It allows you to define the data limit that a query call can render, thus enabling you to paginate the data and determine the pagination boundaries.
 
 Follow the steps below to paginate the responses and request smaller chunks of data at a time:
 
-1. Enable the Server Side Pagination property for the List.
-2. Call the query on the onPageChange event listener.
-3. Set the LIMIT and OFFSET clause in the query:
+1. Enable the **Server Side Pagination** property for the List.
+2. Call the query on the **onPageChange** event listener.
+3. Set the `LIMIT` and `OFFSET` clause in the query:
```

Comment:
Rather than saying set limit and offset, we should say you can use <ListName>.pageSize for the limit and (<ListName>.pageNo-1)*pageSize as offset. 

@ashit-rath @Tooluloope we should also add a pageOffset field which calculates offset similar to table

Reply:
updated